test/unit/controller.js

```diff
@@ -894,7 +905,9 @@ exports['controller.root'] = {
 
     this.child = new FakeChild();
 
-    this.spawn = this.sandbox.stub(cp, 'spawn', () => this.child);
+    this.spawn = this.sandbox.stub(cp, 'spawn', () => {
+      return this.child;
+    });
```

Comment:
No, I had to do some delicate rebasing with the first version of this branch and your self/bind branch. No big deal though, I will fix this